client/app/scripts/superdesk-search/tests/search_spec.js

```diff
@@ -21,6 +21,59 @@ describe('search service', function() {
         expect(criteria.query.filtered.query.query_string.query).toBe('test');
     }));
 
+    it('can create query for from_desk and to_desk', inject(function($rootScope, search) {
+        // only from desk is specified
+        var criteria = search.query({from_desk: 'test-authoring'}).getCriteria();
+        var filters = criteria.query.filtered.filter.and;
+        expect(filters).toContain({term: {'task.last_authoring_desk': 'test'}});
+        criteria = search.query({from_desk: 'test-production'}).getCriteria();
+        filters = criteria.query.filtered.filter.and;
+        expect(filters).toContain({term: {'task.last_production_desk': 'test'}});
+
+        // only to desk is specified
+        criteria = search.query({to_desk: '456-authoring'}).getCriteria();
+        filters = criteria.query.filtered.filter.and;
+        expect(filters).toContain({term: {'task.desk': '456'}});
+        expect(filters).toContain({exists: {field: 'task.last_production_desk'}});
+        criteria = search.query({to_desk: '456-production'}).getCriteria();
+        filters = criteria.query.filtered.filter.and;
+        expect(filters).toContain({term: {'task.desk': '456'}});
+        expect(filters).toContain({exists: {field: 'task.last_authoring_desk'}});
+
+        // both from desk and to desk are specified
+        criteria = search.query({from_desk: '123-authoring', to_desk: '456-production'}).getCriteria();
+        filters = criteria.query.filtered.filter.and;
+        expect(filters).toContain({term: {'task.last_authoring_desk': '123'}});
+        expect(filters).toContain({term: {'task.desk': '456'}});
+    }));
```

Comment:
I would recommend splitting this into three separate test cases, with each covering only a single use case - more specific error report if something goes wrong.